hazelcast/src/main/java/com/hazelcast/jet/impl/execution/init/ExecutionPlanBuilder.java

```diff
@@ -63,9 +67,17 @@ private ExecutionPlanBuilder() {
         assert !isLightJob || jobConfig == LIGHT_JOB_CONFIG;
         final JetInstance instance = getJetInstance(nodeEngine);
         final int defaultParallelism = instance.getConfig().getInstanceConfig().getCooperativeThreadCount();
+        final Map<Address, int[]> partitionAssignment = getPartitionAssignment(nodeEngine);
         final Set<MemberInfo> members = new HashSet<>(membersView.size());
-        final Address[] partitionOwners = new Address[nodeEngine.getPartitionService().getPartitionCount()];
-        initPartitionOwnersAndMembers(nodeEngine, membersView, members, partitionOwners);
+        for (Address address : partitionAssignment.keySet()) {
+            MemberInfo member = membersView.getMember(address);
+            if (member == null) {
+                // Address in partition table doesn't exist in member list,
+                // it has just joined the cluster.
+                throw new TopologyChangedException("Topology changed, " + address + " is not in original member list");
+            }
+            members.add(member);
+        }
```

Comment:
this iterates over all the members for each address, since MembersView keeps them in a list.


```suggestion
        Set<Address> membersViewAddresses = membersView.getAddresses();
        for (Address address : partitionAssignment.keySet()) {
            if (!membersViewAddresses.contains(address)) {
                // Address in partition table doesn't exist in member list,
                // it has just joined the cluster.
                throw new TopologyChangedException("Topology changed, " + address + " is not in original member list");
            }
            members.add(member);
        }
```